wpiutil/src/main/native/cpp/leb128.cpp

```diff
@@ -69,7 +69,7 @@ bool ReadUleb128(raw_istream& is, uint64_t* ret) {
       return false;
     }
 
-    result |= (byte & 0x7f) << shift;
+    result |= ((byte & 0x7fULL) << shift) & 0xFFFFFFFF;
```

Comment:
Why the mask?  It's a 64-bit return value, the mask will limit it to 32-bit.

Reply:
The test fails if I don't. Maybe our impl of ULEB128 is wrong?

Reply:
It looks like the last SizeULEB test is wrong… it should be size 6, not size 5.

Reply:
The reference impl on [Wikipedia](https://en.wikipedia.org/wiki/LEB128#Encode_unsigned_integer) also gives 5 for UINT32_MAX. The encoded string is "0xff 0xff 0xff 0xff 0x0f".

Reply:
Whoops, my bad.  Yeah, 32 bits fits in 5 bytes (5*7=35 bits).  What tests are failing?

Reply:
```
[ RUN      ] LEB128Test.ReadUleb128
 result=0
   byte=0
 lshift=0
shifted=0

 result=0
   byte=1
 lshift=0
shifted=1

 result=0
   byte=111111
 lshift=0
shifted=111111

 result=0
   byte=1000000
 lshift=0
shifted=1000000

 result=0
   byte=1111111
 lshift=0
shifted=1111111

 result=0
   byte=0
 lshift=0
shifted=0

 result=0
   byte=1
 lshift=7
shifted=10000000

 result=0
   byte=1
 lshift=0
shifted=1

 result=1
   byte=1
 lshift=7
shifted=10000000

 result=0
   byte=10000
 lshift=0
shifted=10000

 result=16
   byte=1
 lshift=7
shifted=10000000

 result=0
   byte=1111111
 lshift=0
shifted=1111111

 result=127
   byte=1
 lshift=7
shifted=10000000

 result=0
   byte=0
 lshift=0
shifted=0

 result=0
   byte=10
 lshift=7
shifted=100000000

 result=0
   byte=1
 lshift=0
shifted=1

 result=1
   byte=10
 lshift=7
shifted=100000000

 result=0
   byte=0
 lshift=0
shifted=0

 result=0
   byte=1000001
 lshift=7
shifted=10000010000000

 result=8320
   byte=0
 lshift=14
shifted=0

 result=8320
   byte=0
 lshift=21
shifted=0

 result=8320
   byte=10000
 lshift=28
shifted=100000000000000000000000000000000

/home/tav/git/wpilib/allwpilib/wpiutil/src/test/native/cpp/leb128Test.cpp:74: Failure
Expected equality of these values:
  8320u
    Which is: 8320
  val
    Which is: 4294975616
```
I tried adding the "read" test for 8320u to the "write" tests, and it fails going the other way too.
```
Expected equality of these values:
  expected
    Which is: "\x80\xC1\x80\x80\x10"
  buf.str()
    Which is: "\x80" "A"
```